libs/shared/components/src/Buttons/ChainButton.tsx

```diff
@@ -0,0 +1,44 @@
+import { Button, Typography } from '@mui/material';
+import { arbitrum } from 'viem/chains';
+
+import { ChainIcon } from '../Icons';
+
+import type { ButtonProps } from '@mui/material';
+import type { OverridableComponent } from '@mui/material/OverridableComponent';
+import type { TypographyTypeMap } from '@mui/material/Typography/Typography';
+import type { Chain } from 'viem';
+
+import type { ChainIconProps } from '../Icons';
+
+export type ChainButtonProps = {
+  chain: Chain;
+  iconProps?: Partial<ChainIconProps>;
+  labelProps?: Partial<OverridableComponent<TypographyTypeMap>>;
+  isDisabled?: boolean;
+} & ButtonProps;
+
+export const ChainButton = ({
+  chain,
+  iconProps,
+  labelProps,
+  isDisabled,
+  ...rest
+}: ChainButtonProps) => {
+  return (
+    <Button
+      {...rest}
+      sx={{
+        display: 'flex',
+        gap: 1,
+        alignItems: 'center',
+        ...rest?.sx,
+        paddingLeft: 1,
+      }}
+    >
+      <ChainIcon chainId={chain.id} sx={{ height: '28px' }} {...iconProps} />
```

Comment:
tried this and it doesn't look good, so keeping as is

Reply:
Just confused why you're against using a height if I know exactly what height I want my icon to be.

Reply:
because setting `height: 28px` only actually gives you a 24x24 px icon since the width is constrained, your icon is not 28px tall, but 24. If you set the fontSize, then it will be 28x28